modin/engines/base/io/text/csv_dispatcher.py

```diff
@@ -181,6 +222,10 @@ def _read(cls, filepath_or_buffer, **kwargs):
             new_index = pandas.RangeIndex(sum(row_lengths))
         else:
             index_objs = cls.materialize(index_ids)
+            index_objs = [
+                pandas.Index([]) if isinstance(obj, int) and obj == 0 else obj
+                for obj in index_objs
+            ]
```

Comment:
in which cases integers appear among index objects?

Reply:
It occurs when `len(names)` is lower than actual number of columns in file and `skiprows` makes DataFrames emty in some partitions (for example if `skiprows` is `lambda x: x > 25`). in modin tests this case corresponds, for example, to `modin/pandas/test/test_io.py::TestCsv::test_read_csv_parsing_2[None-names0-None-<lambda>-0] ` case
In this case some of the last partitions will contain empty DataFrames with index [], which then will be converted to zero by getting the length of this list.

Reply:
I think in that case we can just skip partition instead of mixing Index objects and integers, look at [this](https://github.com/modin-project/modin/pull/2607/files#r574306355) comment

Reply:
Yes, you are right, `index_objs` modification was removed.